romanesco/format/__init__.py

```diff
@@ -54,17 +74,109 @@ def csv_to_rows(input):
 
     return output
 
-converters = {}
-validators = {}
+
+def converter_path(source, target):
+    """Gives the shortest path that should be taken to go from a source
+    type/format to a target type/format.
+
+    Throws a ``NetworkXNoPath`` exception if it can not find a path.
+
+    :param source: Validator tuple indicating the type/format being converted `from`.
+    :param target: ``Validator`` tuple indicating the type/format being converted `to`.
+    :returns: An ordered list of the analyses that need to be run to convert from
+    ``source`` to ``target``.
+    """
+    # These are to ensure an exception gets thrown if source/target don't exist
+    get_validator(source)
+    get_validator(target)
+
+    # We sort and pick the first of the shortest paths just to produce a stable
+    # conversion path. This is stable in regards to which plugins are loaded at the
+    # time.
+    paths = all_shortest_paths(conv_graph, source, target)
+    path = sorted(paths)[0]
+    path = zip(path[:-1], path[1:])
+
+    return [get_edge(u, v)[2] for (u, v) in path]
```

Comment:
We have the to and from ```Validator``` objects but not the associated analysis, that is stored in the edge data. ```NetworkX``` edges are formatted as ```(u, v, data)``` so ```get_edge(u, v)[2]``` is just returning the analysis. I've cleaned this up and documented it in d36400a